### PR TITLE
Change that seems to enable EBPF back end to build on Ubuntu 16.04

### PR DESCRIPTION
**Pull request by jafingerhut**

### CODE REVIEW

**Comment by jafingerhut** [comment]
This is just an experimental change I tried that seems to enable the EBPF back end to build on a freshly installed Ubuntu 16.04 Linux system.  Don't trust that I know what I'm doing here :)

**Comment by mbudiu-vmw** [comment]
Does this enable the kernel-based tests?
It's quite easy to write a condition to disable them, what we want is to be able to run them on systems that support it.

File: backends/ebpf/CMakeLists.txt [hunk 1]
```diff
@@ -56,8 +56,8 @@ execute_process(COMMAND uname -r
   OUTPUT_STRIP_TRAILING_WHITESPACE
   RESULT_VARIABLE rc)
 message(STATUS "Detected kernel version: ${P4C_EBPF_KERNEL_VER}")
-if (EXISTS /usr/src/linux-headers-${P4C_EBPF_KERNEL_VER}/include/uapi/linux)
-  include_directories(/usr/src/linux-headers-${P4C_EBPF_KERNEL_VER}/include/uapi/linux)
+if (EXISTS /usr/src/linux-headers-${P4C_EBPF_KERNEL_VER}/include)
+  include_directories(/usr/src/linux-headers-${P4C_EBPF_KERNEL_VER}/include)
```
Comment (cc10512) [inline backends/ebpf/CMakeLists.txt]: I'm afraid this is not going to include the right header that EBPF needs.
Can out try:
```if (EXISTS /usr/src/linux-headers-${P4C_EBPF_KERNEL_VER}/include)
  include_directories(/usr/src/linux-headers-${P4C_EBPF_KERNEL_VER}/include /usr/src/linux-headers-${P4C_EBPF_KERNEL_VER}/include/uapi/linux)
```

Comment (mbudiu-vmw) [reply backends/ebpf/CMakeLists.txt]: What is the error? The compilation should never fail, running the tests may fail.

**Comment by jafingerhut** [comment]
With the alternate change in the attached file patch1.txt suggested by Calin, on a freshly installed Ubuntu 16.04 Linux system, plus all dependencies recommended in the p4c README's installed, I get the error message in the attached file patch1-build-error-msg.txt

[patch1.txt](https://github.com/p4lang/p4c/files/3708666/patch1.txt)
[patch1-build-error-msg.txt](https://github.com/p4lang/p4c/files/3708667/patch1-build-error-msg.txt)

File: backends/ebpf/CMakeLists.txt (hunk 1 above)
Comment (jafingerhut) [reply backends/ebpf/CMakeLists.txt]: See this comment on this same PR: https://github.com/p4lang/p4c/pull/2049#issuecomment-540106445
for attached files with what I think is a slightly different change suggested by Calin, and the error message I see when I attempt to build p4c with that different change.

**Comment by mbudiu-vmw** [comment]
Hm, we should not set include_directories for gcc, only for the tests.

**Comment by jafingerhut** [comment]
If you have a suggestion for how to change the p4c repo to set include directories for tests, not for gcc, I can try it out.  I don't know how to do that.

**Comment by mbudiu-vmw** [comment]
I think we just have to take this line out:
`include_directories(/usr/src/linux-headers-${P4C_EBPF_KERNEL_VER}/include/uapi/linux)`
We have to do something equivalent in the python scripts which compile the resulting C files, though. I will work on it.

**Comment by jafingerhut** [comment]
I have pushed another commit on this PR so that I believe it follows Mihai's suggestion of his previous comment -- removing the include_directories line.

**Comment by jafingerhut** [comment]
I have confirmed that with the change proposed in the latest version of this PR (the one that only deletes one line, an `include_directories` directive), the latest version of p4c source code builds without errors on both a freshly installed & updated Ubuntu 16.04 system, and Ubuntu 18.04.

**Comment by mbudiu-vmw** [comment]
Let's merge this and we'll have to do something more to make the ebpf tests properly compile on some architectures.

**Comment by cc10512** [comment]
Does it also pass the tests?

**Comment by mbudiu-vmw** [comment]
At least on Travis...

**Comment by jafingerhut** [comment]
@cc10512 @mbudiu-vmw It also passed all p4c tests _except_ the ebpf-specific ones on my freshly installed Ubuntu 16.04 and 18.04 machines.  We shouldn't be surprised about failing the ebpf-specific ones given the lack of changes that Mihai is planning to make, and that I have done nothing special to install anything that the ebpf back end might need, except llvm.

**Pull request merged**